controlplane/internal/imageregistry/image_registry.go

```diff
@@ -142,36 +148,43 @@ func parseMirrorRegistries(
 	mirrorTrees []*toml.Tree,
 	insecureRegistries *[]string,
 	source string,
-) {
-	itmsMirror := configv1.ImageTagMirrors{}
-	idmsMirror := configv1.ImageDigestMirrors{}
+) error {
+	itmsMirror := &configv1.ImageTagMirrors{}
+	idmsMirror := &configv1.ImageDigestMirrors{}
 	for _, mirrorTree := range mirrorTrees {
-		if mirror, ok := mirrorTree.Get("location").(string); ok {
-			if insecure, ok := mirrorTree.Get("insecure").(bool); ok && insecure {
-				*insecureRegistries = append(*insecureRegistries, mirror)
-			}
-			if pullFrom, ok := mirrorTree.Get("pull-from-mirror").(string); ok {
-				switch pullFrom {
-				case "tag-only":
-					itmsMirror.Source = source
-					itmsMirror.Mirrors = append(itmsMirror.Mirrors, configv1.ImageMirror(mirror))
-				case "digest-only":
-					idmsMirror.Source = source
-					idmsMirror.Mirrors = append(idmsMirror.Mirrors, configv1.ImageMirror(mirror))
-				}
-			} else {
-				// Default is pulling by digest
-				idmsMirror.Source = source
-				idmsMirror.Mirrors = append(idmsMirror.Mirrors, configv1.ImageMirror(mirror))
-			}
+		mirror, ok := mirrorTree.Get("location").(string)
+		if !ok {
+			return fmt.Errorf("missing mirror registry location key or the value of location is not a string")
+		}
+		if insecure, ok := mirrorTree.Get("insecure").(bool); ok && insecure {
+			*insecureRegistries = append(*insecureRegistries, mirror)
 		}
+		setPullPolicy(itmsMirror, idmsMirror, source, mirror, mirrorTree)
```

Comment:
we can already know that the mirror is going to be `configv1.ImageMirror(mirror)`, also that itmsMirror.Source and idmsMirror.Source is `source`. We will then need to decide where to add the mirror: will it be itms or idms? 

```golang
mirror, isInsecure := getMirrorFromToml(mirrorTree)
imageMirror := configv1.ImageMirror(mirror)
if pullFrom, ok := mirrorTree.Get("pull-from-mirror").(string); ok && pullFrom == "tag-only" {
    itmsMirrors.Mirrors = append(itmsMirrors.Mirrors, imageMirror)
    continue
}
idmsMirrors.Mirrors = append(idmsMirrors.Mirrors, imageMirror)
```

Also I think we can reorganize the `getImageRegistries` function, maybe using a specific struct to encapsulate values from current `parse` function (that will need renaming). For example:
```golang
type RegistryConfig struct {
    TagMirrors *configv1.ImageTagMirrors{}
    DigestMirrors *configv1.ImageDigestMirrors{}
    InsecureRegistries []string
}

Reply:
Good idea! Modified it, how does it look?